### ISSUE TITLE
fallback to default page for i18n

### ISSUE THREAD

**Issue by alirezaalavi87** [Keats/gutenberg#2659]
# Bug Report
for every language your site supports, for every page you MUST create a separate file for that language (https://www.getzola.org/documentation/content/multilingual/)

This leads to many redundant pages in some cases. 

I purpose if for a language there is no `index.{code}.md` zola should just fallback to `index.md`, and maybe show a warning instead of not creating your website at all.

for example see [duckquill](https://codeberg.org/daudix/duckquill) and how many redundant pages it has to have in itslef for no reason.

## Environment

Zola version:
v0.19.2

## Expected Behavior

I purpose if for a language there is no `index.{code}.md` zola should just fallback to `index.md`, and maybe show a warning instead of not creating the website at all.

**Comment by Keats** [comment]
> for every language your site supports, for every page you MUST create a separate file for that language

That's not true? Only the sections require multiple files if you want them, not the pages. As for fallback, I don't think we can do it? The slug in the url might be different

**Comment by alirezaalavi87** [comment]
yes my bad, for every section we will need a new page. but still I think a fallback would be very convenient. because maybe I don't want to change some sections  based on language and maintain two copies of the same page.

@Keats

**Comment by alirezaalavi87** [comment]
And for another example, checkout the gallery page in my website.[link](https://alavi.me/gallery)

I want the posts in my gallery to also be loaded in other languages, without copying about a hundred `index.md` files and renaming them to `index.fa.md`.

I'm not sure if this has another workaround or not, but I think the point is valid.

I think the fallback behavior should be in a way that you can include or exclude pages or sections to use fallback in `config.default_language`.

for example, I want all my gallery posts to fallback to `index.md` but not my blog posts. I want only blog posts with `index.fa.md` when I'm in farsi mode and `index.md` in english mode.
 
I think this is a reasonable behavior.

**Comment by alirezaalavi87** [comment]
bump?

**Comment by Keats** [comment]
How would that look like for a user? Zola would automatically generate a page for every non-default language that redirects to the default language if not present?

**Comment by alirezaalavi87** [comment]
@Keats 
In my opinion, the default fallback should behave like this:

I have multiple posts in my gallery. I want those with `index.fa.md` to be rendered and those without it, just fallback to `index.md`. so zola has to render `index.fa.md` itself so we can have:

```
gallery/
    post-1/
        index.md
    post-2/
        index.md
        index.fa.md
```
and render

`test.com/fa/gallery/post-1`
`test.com/fa/gallery/post-2`

remember that whether we should do fallbacks to `index.md` or not should be a per-section option. so I want to fallback in my `gallery` but not in my `blog`

**Comment by Keats** [comment]
Seems ok to me

**Comment by 0xis-cn** [comment]
I wonder whether fallback or *supporting translingual section/taxonomy* is better. When I am determined, I am happy to create a pull request.

**Comment by Keats** [comment]
> supporting translingual section/taxonomy is better

It would be nice but i think an optional fallback option would be required either way
